Add support for private ExtendedKey for conversion to ExtendedPublicKey

pezcore opened the pull request:
Add support for converting private `ExtendedKey` to `ExtendedPublickKey<K>` via `TryFrom

tony-iqlusion commented on bip32/src/extended_key/public_key.rs:
```diff
@@ -135,8 +135,83 @@ where
                 public_key: PublicKey::from_bytes(extended_key.key_bytes)?,
                 attrs: extended_key.attrs.clone(),
             })
+        } else if extended_key.prefix.is_private() {
```
You'll need to gate this (as well as the `XPrv` import) on the `secp256k1` feature being available:

```suggestion
        } else if cfg!(feature = "k256") && extended_key.prefix.is_private() {
```

pezcore replied:
How can I feature-gate the `XPrv` import? If `use crate::XPrv` is feature-gated, then this code will not compile when `--no-default-features` because `XPrv` is still referenced on the line below.

tony-iqlusion replied:
You can either do:

```rust
#[cfg(feature = "k256")]
use crate::XPrv;
```

or refer to it directly as `crate::XPrv`

tony-iqlusion commented:
@pezcore some build failures related to unused constants that need to be feature-gated

pezcore commented:
@tony-iqlusion  ok I think I fixed all the unit tests that were failing due to the various features combinations. Could you take a look at how I implemented the patch w.r.t features and offer some feedback though? I wasn't really sure how to go about doing this and the current implementation feels very non-idiomatic.

tony-iqlusion commented:
Looks like it needs rustfmt.

Don't worry about the coverage check. The breakage there looks unrelated.

pezcore commented:
ok I fixed the `rustfmt` issue.
